### ISSUE TITLE
VidLanKD: Improving Language Understanding via Video-Distilled Knowledge Transfer

### ISSUE THREAD

**Issue by reyllama** [reyllama/paper-reviews#133]
## Motivation

- Language models can do better with the understanding of the visual world

## TL; DR

- Pretrain a teacher model with two objectives: MLM and Video/Caption contrastive learning (triplet loss)
- Apply KD to the student network (identical architecture as teacher language model, Transformer) with NST and CRD objective

## Previous Work

- Vokenization: employ image level visual signals for better language modeling (on top of BERT)

## Method

![image](https://user-images.githubusercontent.com/46308407/145174677-b713f279-9134-4932-a00f-b02bb30c4d01.png)

- Teacher model consists of **language encoder** and **video encoder**. 
- For language encoder, BERT-like transformer encoder is used.
- For video encoder, 2D image encoder (CLIP gave the best performance) and 3D video encoder (3D-ResNeXT-152) was used.
- Video encoder was frozen during training. On top of pretrained video encoder, a Transformer based model was added to yield the final video-level representation (temporal features were simply average pooled)
- Two objectives: video-text contrastive learning and masked language modeling

![image](https://user-images.githubusercontent.com/46308407/145175381-9ace3a8e-ffef-45e8-bc0c-73ca956e21ae.png)

![image](https://user-images.githubusercontent.com/46308407/145175091-08614cac-69c1-4fcc-9ba7-1dd67b5d9147.png)
- The student model is trained on two objectives: masked language modeling and knowledge distillation.
- For knowledge distillation, explored five representatives: soft labels (cross-entropy), L2 regression (feature), Neuron Selectivity Transfer (NST), Contrastive Representation Distillation (CRD) and Vokenization (predicting the retrieved video index).
- Since NST and CRD combined gave the best result, they used these two objectives for KD.

![image](https://user-images.githubusercontent.com/46308407/145175415-15291e42-141f-458f-bf7f-47c3313a1822.png)

- NST: compute MMD (Maximum Mean Discrepancy) between hidden states of teacher and students using Gaussian Kernel
- CRD: maximizes mutual information using Info-NCE-like loss function (NT-Xent)

### Dataset
- Video/Text: HowTo100M
- Text: English Wikipedia

### Evaluations

![image](https://user-images.githubusercontent.com/46308407/145175784-0673fa02-8819-4185-8b90-16c0cfc59c1c.png)
![image](https://user-images.githubusercontent.com/46308407/145175883-d13f53b1-ea94-4ab5-a802-41b33752c92a.png)

- Text-only pretraining on HowTo100M gives inferior performance -> Multimodal training matters, not the dataset(HowTo100M, which is bigger in size)

![image](https://user-images.githubusercontent.com/46308407/145175987-d1e5a25f-62fe-48f7-a07b-c7916b4a3935.png)
- Both MLM and Contrastive objective contribute to the performance.

![image](https://user-images.githubusercontent.com/46308407/145176093-e3aec686-f402-4a0c-acf1-bbb8b04ad7f5.png)
- KD works better than 2-stage pretraining with a single model

![image](https://user-images.githubusercontent.com/46308407/145176132-efb566f9-4d02-4ded-870d-2b5e082233c3.png)
- NST + CRD gives the best performance out of 5 KD objectives.

![image](https://user-images.githubusercontent.com/46308407/145176224-f7c2a1e5-74e0-4ded-bf8f-06614270fb70.png)

![image](https://user-images.githubusercontent.com/46308407/145176305-7dcf1237-627f-4921-999e-92bc0645d2a3.png)
![image](https://user-images.githubusercontent.com/46308407/145176426-cceeb14b-be28-4054-a952-7e15ab6a325b.png)

## Comments

- Multi-modal pretraining is expected to give better performance compared to text-only pretraining. The important part is how to integrate signals from different modalities to currently well-functioning language models.
- This is rather simpler compared to the vokenization paper (EMNLP'20), but the result is more clear and interpretable thanks to extensive ablation studies.
- By simply integrating video-text contrastive objective, the language model can better reason in diverse benchmarks, which is interesting.
- Also, we observed that simply training a single model with two pretraining stages (text+video & text-only) doesn't match the performance of KD. Wonder what's the mystery behind knowledge distillation (and self-distillation :) )
